Alignments of Data Model Section with SD-JWT and SD-JWT-VC latest drafts (#511)
italia/eidas-it-wallet-docs: docs/en/pid-eaa-data-model.rst

In case of nested object in a SD-JWT payload each claim, on each level of the JSON, should be individually selectively disclosable or not. Therefore **_sd** claim containing digests MAY appear multiple times at different level in the SD-JWT.

For each claim that is an array element the digests of the respective disclosures and decoy digests are added to the array in the same position of the original claim values as specified in Section 4.2.4.2 of `SD-JWT`_.

In case of array elements, digest values are calculated using a hash function over the disclosures, containing:

  - a random salt, 
  - the array element

> In case of multiple array elements, the Issuer may wish to conceal presence of any statement while also allowing the Holder to reveal each of those elements individually (Section 4.2.6 `SD-JWT`_). Both the entire array and the individuals entries can be selective disclosure.

Reviewer: ```suggestion
In case of multiple array elements, the Issuer may hide the value of the entire array or any of the entry contained within the array, the Holder can disclose both the entire array and any single entry within the array, as defined in Section 4.2.6 of `SD-JWT`_.
```

      - [NSD].The value MUST be an "integrity metadata" string as defined in Section 3 of [`W3C-SRI`_]. *SHA-256*, *SHA-384* and *SHA-512* MUST be supported as cryptographic hash functions. *MD5* and *SHA-1* MUST NOT be used. This claim MUST be verified according to Section 3.3.5 of [`W3C-SRI`_].
      - Section 6.1 `SD-JWT-VC`_, [`W3C-SRI`_]
    * - **verification**
      - [SD]. Object containing user authentication and user data verification information. It is REQUIRED only if the Credential type is PersonIdentificationData, and it MUST not be included in other Credential types. It MUST include the following sub-value:

>           * ``assurance_level``: String identifying the level of identity assurance guaranteed during the User authentication process.

Reviewer 2: Not clear, considering the PersonIdentificationData, are there cases where the identity assurance level during the User authentication process is not high?

      - [NSD].The value MUST be an "integrity metadata" string as defined in Section 3 of [`W3C-SRI`_]. *SHA-256*, *SHA-384* and *SHA-512* MUST be supported as cryptographic hash functions. *MD5* and *SHA-1* MUST NOT be used. This claim MUST be verified according to Section 3.3.5 of [`W3C-SRI`_].
      - Section 6.1 `SD-JWT-VC`_, [`W3C-SRI`_]
    * - **verification**

>       - [SD]. Object containing user authentication and user data verification information. It is REQUIRED only if the Credential type is PersonIdentificationData, and it MUST not be included in other Credential types. It MUST include the following sub-value:

Reviewer: ```suggestion
      - [SD]. Object containing User authentication and User data verification information. When the Credential type is set to `PersonIdentificationData`, the `verification` claim MUST be included by the Issuer. When present, the `verification` claim MUST be set with a JSON Object including the following members:
```

          * ``assurance_level``: String identifying the level of identity assurance guaranteed during the User authentication process.

>           * ``evidence``: Each entry of the array MUST contain the following sub-values:

Reviewer: ```suggestion
          * ``evidence``: Each entry of the array MUST contain the following members:
```

>               - ``type``: It represents evidence type. IT MUST be set to ``vouch``.

Reviewer: ```suggestion
              - ``type``: It represents evidence type. It MUST be set to ``vouch``.
```

              - ``time``: UNIX Timestamps with the time of the authentication or verification.

>               - ``attestation``: It MUST contain the following claims:

Reviewer: ```suggestion
              - ``attestation``: It MUST contain the following members:
```

    * - **schema_uri**
      - CONDITIONAL. REQUIRED if **schema** is not present.
      - [`SD-JWT-VC`_] Section 6.2.
    * - **schema_uri#integrity**
      - CONDITIONAL. REQUIRED if **schema_uri** is present.
      - [`SD-JWT-VC`_] Section 6.2.
    * - **data_source**
      - REQUIRED. Object containing information about the data origin. It MUST contain the object ``verification`` with this following sub-value:

>           * ``trust_framework``: MUST contain trust framework used for digital authentication towards authentic source system.
>           * ``authentic_source``: MUST contain the following claims related to information about the authentic source:
>                * ``organization_name`` name of the authentic source.
>                * ``organization_code`` code identifier of the authentic source.
>                * ``homepage_uri`` uri pointing to the authentic source's homepage.
>                * ``contacts`` contact list for info and assistance.
>                * ``logo_uri`` URI pointing to the logo image.

Reviewer: ```suggestion
          * ``trust_framework``: MUST contain trust framework used for digital authentication towards Authentic Source system.
          * ``authentic_source``: MUST contain the following claims related to information about the Authentic Source:
               * ``organization_name`` name of the Authentic Source.
               * ``organization_code`` code identifier of the Authentic Source.
               * ``homepage_uri`` uri pointing to the Authentic Source's homepage.
               * ``contacts`` contact list for info and assistance.
               * ``logo_uri`` URI pointing to the logo image.
```

      - This specification
    * - **display**
      - REQUIRED. Array of objects, one for each language supported, containing display information for the Digital Credential type. It contains for each object the following properties:

          * ``lang``: language tag as defined in :rfc:`5646` Section 2. [REQUIRED].
          * ``name``: human-readable label for the Digital Credential type. [REQUIRED].
          * ``description``: human-readable description for the Digital Credential type. [REQUIRED].

>           * ``rendering``: object containing rendering methods supported by the Digital Credential type. [REQUIRED].
>           Rendering method `svg_template` MUST be supported.

Reviewer: ```suggestion
          * ``rendering``: object containing rendering methods supported by the Digital Credential type. [REQUIRED]. The rendering method `svg_template` MUST be supported.
```

           If rendering method `simple` is also supported, the ``simple`` object contains the following properties: 
             * ``logo``: object containing information about the logo to display. This property is REQUIRED. The object contains the following sub-values:
                * ``uri``: URI pointing to the logo image. [REQUIRED]
                * ``uri#integrity``: integrity metadata as defined in Section 3 of `W3C-SRI`_. [REQUIRED].
                * ``alt_text``: A string containing alternative text to display instead of the logo image. [OPTIONAL].
             * ``background_color``: RGB color value as defined in `W3C.CSS-COLOR`_ for the background of the Digital Credential. [OPTIONAL].
             * ``text_color``: RGB color value as defined in `W3C.CSS-COLOR`_ for the text of the Digital Credential. [OPTIONAL].
          .. note::
            The use of the SVG template is recommended for all applications that support it.
      - [`SD-JWT-VC`_] Section 8.
    * - **claims**

>       - REQUIRED. Array of objects containing information for displaying and validating Digital credential claims. It contains for each credential claim the following properties:

Reviewer: ```suggestion
      - REQUIRED. Array of objects containing information for displaying and validating Digital Credential claims. It contains for each Credential claim the following properties:
```

          * ``path``: array indicating the claim or claims that are being addressed. [REQUIRED].

>           * ``display``: array containing display information about the claim indicated in the ``path``. The array contains an object for each language supported by the Digital Credential type. This property is REQUIRED. It contains the following sub-values:

Reviewer: ```suggestion
          * ``display``: array containing display information about the claim indicated in the ``path``. The array contains an object for each language supported by the Digital Credential type. This property is REQUIRED. It contains the following members:
```
